cadccutout/cadccutout/file_helpers/fits/fits_file_helper.py

```diff
@@ -148,17 +135,56 @@ def _post_sanitize_header(self, header, cutout_result):
                     cd_key = 'CD{}'.format(idx_val)
                     cd_val = header.get(cd_key)
 
-                    if cd_val is not None:
+                    if cd_val:
                         pc_key = 'PC{}'.format(idx_val)
-                        if header.get(pc_key) is None:
-                            header.set(pc_key, cd_val)
 
-                        header.remove(
-                            cd_key, ignore_missing=True, remove_all=True)
+                        if not header.get(pc_key):
+                            header.rename_keyword(cd_key, pc_key)
+                        else:
+                            header.remove(cd_key)
+
+                        if i == 0 and j == 0:
+                            pc11_key_idx = header.index(pc_key)
+
+            # Check for padded (leading) zeroes in the PC key names.
+            for i in range(naxis):
+                for j in range(naxis):
+                    pc_key = 'PC0{}_0{}'.format(i + 1, j + 1)
+                    pc_val = header.get(pc_key)
+
+                    if pc_val or pc_val == 0:
```

Comment:
This is just if pc_val not None?

Reply:
Right.  I take it back, `cd_val` can be zero.  Changed.